diff --git a/ext/imagick/config.m4 b/ext/imagick/config.m4
index 441fc8b..a81609c 100644
--- a/ext/imagick/config.m4
+++ b/ext/imagick/config.m4
@@ -15,7 +15,8 @@ m4_include(ifdef('PHP_IMAGICK_STATIC',PHP_EXT_BUILDDIR(imagick)[/],)[imagemagick
 
 # This line checks that ImageMagick is available, and
 # meets our minimum supported version. TODO change to 6.7.0
-IM_FIND_IMAGEMAGICK([6.2.4], [$PHP_IMAGICK])
+# @Wasmer: skip this, we provide the _LIBS and _CFLAGS vars manually
+# IM_FIND_IMAGEMAGICK([6.2.4], [$PHP_IMAGICK])
 
 #
 # PHP minimum version
diff --git a/ext/imagick/php_imagick.h b/ext/imagick/php_imagick.h
index 2bfda78..c966903 100644
--- a/ext/imagick/php_imagick.h
+++ b/ext/imagick/php_imagick.h
@@ -25,7 +25,7 @@
 #define PHP_IMAGICK_EXTNAME    "imagick"
 // The version is deliberately left as 'PACKAGE_VERSION' in source code.
 // It is only replaced with the actual version number that packaged through pecl.php.net
-#define PHP_IMAGICK_VERSION    "@PACKAGE_VERSION@"
+#define PHP_IMAGICK_VERSION    "3.7.0"
 #define PHP_IMAGICK_EXTNUM     30700
 
 /* Import configure options when building 
