diff --git a/crates/cli/src/lib.rs b/crates/cli/src/lib.rs
--- a/crates/cli/src/lib.rs
+++ b/crates/cli/src/lib.rs
@@ -1 +1,2 @@
+#![cfg_attr(target_os = "wasi", feature(wasi_ext))]
 /*!
@@ -243 +243,40 @@ pub fn is_readable_stdin() -> bool {
-    #[cfg(not(any(unix, windows)))]
+    #[cfg(target_os = "wasi")]
+    fn imp() -> bool {
+        use std::{
+            fs::File,
+            os::{
+                fd::{FromRawFd, IntoRawFd},
+                wasi::fs::FileTypeExt,
+            },
+        };
+
+        let file = unsafe { File::from_raw_fd(0) };
+        let metadata = file.metadata();
+        let _ = file.into_raw_fd();
+        let md = match metadata {
+            Ok(md) => md,
+            Err(err) => {
+                log::debug!(
+                    "for heuristic stdin detection on WASI, \
+                     could not get file metadata for stdin \
+                     (thus assuming stdin is not readable): {err}",
+                );
+                return false;
+            }
+        };
+        let ft = md.file_type();
+        let is_file = ft.is_file();
+        let is_socket_stream = ft.is_socket_stream();
+        let is_socket_dgram = ft.is_socket_dgram();
+        let is_readable = is_file || is_socket_stream || is_socket_dgram;
+        log::debug!(
+            "for heuristic stdin detection on WASI, \
+             found that is_file={is_file}, \
+             is_socket_stream={is_socket_stream} and \
+             is_socket_dgram={is_socket_dgram}, \
+             and thus concluded that is_stdin_readable={is_readable}",
+        );
+        is_readable
+    }
+
+    #[cfg(not(any(unix, windows, target_os = "wasi")))]
